### DISCUSSION TITLE
Gamecube Logitech Speed Force FFB

### DISCUSSION THREAD

**Discussion by starkninja** [darthcloud/BlueRetro#549] [Q&A]
I am trying to get force feedback working through the BlueRetro GC adapter. The controller is detected properly and the inputs function, just without FFB.

Tried so far:
- Updated to the latest firmware (1.8.2_hw2)
- Tried both normal boot and passive boot (holding down reset)
- Adjusted the outer pins to confirm they are making contact
- Confirmed the wheel is working as expected through the OEM control board

I am using Burnout 2 for testing. The wheel is properly detected as supporting force feedback in the controller settings.

I am available to pull debug information or test beta firmware. I can also test with a multimeter if that would help, but I am not familiar with the expected functionality of those outer connection pins.

Other notes:
The GC is booting Swiss 0.6 from an SD2SP2

**Comment by darthcloud** [comment]
It's not clear to me what you are trying to do.

Are you using a Bluetooth wheel or are you simply trying to use a wired GameCube wheel?

**Reply by starkninja** [reply]
I'm using a wired Logitech wheel. The FFB signal is not being passed through the port, even when booting the Blue Retro in sleep mode.

**Reply by Sauced** [reply]
This may have something to do with the undocumented connection on the GameCube, I asked around during design phases and it was assumed that it was for use with light guns(which never existed) I reused the pins for external programming. 

I did put a jumper in place to enable the original connection in the event that it was actually used, or a new peripheral came out that supported that feature. 
JP2 has a trace between 1 and 2 that will need to be cut, then solder 2 to 3, if this is something that you would prefer we do for you please reach out to our contact page or reply to your confirmation email and I can get you a replacement board with these jumper settings changed. 

![image](https://user-images.githubusercontent.com/3894630/214664018-99152a12-77b3-4ab3-ac4e-559ec61eb70e.png)

**Reply by darthcloud** [reply]
Also just to confirm if you got Bluetooth controller as player 1 and you plug the wheel in port one, does the bluetooth controller move to port 2?

If not maybe the problem is simply the wheel do not connect the shield to ground.

**Reply by starkninja** [reply]
I really appreciate your response, as well as the forward-thinking of your board design. 

I am comfortable changing the jumper and verifying the results.

Just to confirm, will this only disable a hardline programming feature? With no impact to OTA? Sounds reversible either way, just deciding if I should install a simple switch.

I am happy I came across your products. I will be sure to spread the word.

**Reply by darthcloud** [reply]
No impact on OTA

**Reply by Sauced** [reply]
OTA will not be impacted. External serial port programming with a custom cable is the only part that will be affected.

**Reply by starkninja** [reply]
> Also just to confirm if you got Bluetooth controller as player 1 and you plug the wheel in port one, does the bluetooth controller move to port 2?
> 
> If not maybe the problem is simply the wheel do not connect the shield to ground.

Yes, the port shifting functionality is working as expected. The wheel is detected and working properly, just lacking the FFB signal.

The FFB functionality on the Gamecube is quite obsure, and difficult to research. I'll report back once I disconnect that J2 jumper.

**Reply by darthcloud** [reply]
I try to buy one and will documents it :)

**Reply by starkninja** [reply]
> I try to buy one and will documents it :)

I recommend that you try F-Zero GX. There is a patch to add the content from the arcade version (AX).

To confirm my understanding:
![Screenshot_20230125_151037_Gmail.png](https://user-images.githubusercontent.com/18502858/214681009-1ae1c7af-2f78-4e24-b976-fd64ac4c504f.png)

- [ ] desolder at the ❌️
- [ ] bridge the line between pad 2 and pad 3

Sorry for the colors. My options from my phone are limited 🙃

**Reply by Sauced** [reply]
So there are two traces that connect 1-2 they will need to be cut

**Reply by Sauced** [reply]
This is the footprint, the two lines will be coper traces that are built into the board, they just need to be cut.
![image](https://user-images.githubusercontent.com/3894630/214685840-a9371f8f-98d6-4f2f-b2f6-0d683acc4fba.png)
